### PR TITLE
Add endianness conversion for long

### PR DESCRIPTION
**Pull request by AspirinSJL**
... to use in server load reporting PRs.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                            FILE SIZE
 ++++++++++++++ GROWING                                              ++++++++++++++
  +0.0%    +128 [None]                                                  +432  +0.0%
  +0.9%     +32 src/core/lib/iomgr/socket_utils_common_posix.cc          +32  +0.9%
       +13%     +22 [Unmapped]                                               +22   +13%
      [NEW]      +7 grpc_ntohs(unsigned short) [clone .localalias.3]          +7  [NEW]
      [NEW]      +5 grpc_htonl                                                +5  [NEW]
      [NEW]      +5 grpc_ntohl                                                +5  [NEW]

  +0.0%    +160 TOTAL                                                   +464  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                            FILE SIZE
 ++++++++++++++ GROWING                                              ++++++++++++++
  +0.0%    +128 [None]                                                  +432  +0.0%
  +0.9%     +32 src/core/lib/iomgr/socket_utils_common_posix.cc          +32  +0.9%
       +13%     +22 [Unmapped]                                               +22   +13%
      [NEW]      +7 grpc_ntohs(unsigned short) [clone .localalias.3]          +7  [NEW]
      [NEW]      +5 grpc_htonl                                                +5  [NEW]
      [NEW]      +5 grpc_ntohl                                                +5  [NEW]

  +0.0%    +160 TOTAL                                                   +464  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: src/core/lib/iomgr/socket_utils_common_posix.cc [hunk 1]
```diff
@@ -339,6 +339,10 @@ uint16_t grpc_htons(uint16_t hostshort) { return htons(hostshort); }
 
 uint16_t grpc_ntohs(uint16_t netshort) { return ntohs(netshort); }
 
+uint32_t grpc_htonl(uint32_t hostlong) { return htonl(hostlong); }
```
Comment (yashykt) [inline src/core/lib/iomgr/socket_utils_common_posix.cc]: should probably also add something for uv and windows

Comment (nicolasnoble) [reply src/core/lib/iomgr/socket_utils_common_posix.cc]: At that point, honestly, I'd rather we implement our own version of these functions. They are simple enough that they don't warrant the portability hassle.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/core/lib/iomgr/socket_utils_common_posix.cc (hunk 1 above)
Comment (yashykt) [reply src/core/lib/iomgr/socket_utils_common_posix.cc]: given that we already have ntohs functions for all of them, I don't see the harm to adding ntohl functions too.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by AspirinSJL** [comment]
Let's merge this trivial PR first so that I can use it in other PRs. I will add a TODO about re-implementation.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                            FILE SIZE
 ++++++++++++++ GROWING                                              ++++++++++++++
  +0.0%    +128 [None]                                                  +432  +0.0%
  +0.9%     +32 src/core/lib/iomgr/socket_utils_common_posix.cc          +32  +0.9%
       +13%     +22 [Unmapped]                                               +22   +13%
      [NEW]      +7 grpc_ntohs(unsigned short) [clone .localalias.3]          +7  [NEW]
      [NEW]      +5 grpc_htonl                                                +5  [NEW]
      [NEW]      +5 grpc_ntohl                                                +5  [NEW]

  +0.0%    +160 TOTAL                                                   +464  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by yashykt** [comment]
I will defer to Nico's opinion, but it looks good to me.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by AspirinSJL** [comment]
@nicolasnoble Does re-implementing it later sound like a plan? I may need to think about the file arrangement more.

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by nicolasnoble** [comment]
Yes, sure - I was mainly reacting to @yashykt's comment about libuv et al.

**Comment by AspirinSJL** [comment]
- Basic Tests MacOS [opt] (internal CI): #15829
- Interop Cloud-to-Cloud Tests (internal CI): #15831

**Pull request merged**